### PR TITLE
*: Support the statement of "alter user"

### PR DESCRIPTION
**Pull request by zimulala**
Resolving issue #2073 needs to support the statement of "alter user user() ..."

### CODE REVIEW

**Comment by shenli** [comment]
@zimulala Please also update statement metrics. Add a new statement type.

**Comment by zimulala** [comment]
Done @shenli

**Comment by zimulala** [comment]
PTAL @hanfei1991 @coocood  @tiancaiamao

File: executor/executor_simple.go [hunk 1]
```diff
@@ -184,6 +186,60 @@ func (e *SimpleExec) executeCreateUser(s *ast.CreateUserStmt) error {
...
+			User:    user,
+			AuthOpt: s.CurrentAuth,
+		}
+		s.Specs = []*ast.UserSpec{spec}
+	}
+
+	failedUsers := make([]string, 0, len(s.Specs))
+	for _, spec := range s.Specs {
+		userName, host := parseUser(spec.User)
+		exists, err1 := userExists(e.ctx, userName, host)
```
Comment (hanfei1991) [inline executor/executor_simple.go]: Why the name is err1?

File: executor/executor_simple.go [hunk 2]
```diff
@@ -184,6 +186,60 @@ func (e *SimpleExec) executeCreateUser(s *ast.CreateUserStmt) error {
...
+		}
+	}
+
+	err := e.ctx.CommitTxn()
+	if err != nil {
+		return errors.Trace(err)
+	}
+	if len(failedUsers) > 0 {
+		errMsg := "Operation ALTER USER failed for " + strings.Join(failedUsers, ",")
+		return terror.ClassExecutor.New(CodeCannotUser, errMsg)
```
Comment (hanfei1991) [inline executor/executor_simple.go]: Why not use ErrCannotUser.Gen ?

File: executor/executor_simple.go [hunk 3]
```diff
@@ -184,6 +186,60 @@ func (e *SimpleExec) executeCreateUser(s *ast.CreateUserStmt) error {
...
+		userName, host := parseUser(spec.User)
+		exists, err1 := userExists(e.ctx, userName, host)
+		if err1 != nil {
+			return errors.Trace(err1)
+		}
+		if !exists {
+			failedUsers = append(failedUsers, spec.User)
+			if s.IfExists {
+				// TODO: Make this error as a warning.
+			}
```
Comment (hanfei1991) [inline executor/executor_simple.go]: else we should return an error?

Comment (zimulala) [inline executor/executor_simple.go]: No. Add failed users to `failedUsers` enough here. In the end of the function we will return an error.

**Comment by hanfei1991** [comment]
LGTM

File: executor/executor_simple.go (hunk 2 above)
Comment (zimulala) [inline executor/executor_simple.go]: This can wait for you to change in a unified way.

File: executor/executor_simple.go (hunk 3 above)
Comment (tiancaiamao) [inline executor/executor_simple.go]: you can log.Warn here

File: executor/executor_simple.go (hunk 2 above)
Comment (tiancaiamao) [inline executor/executor_simple.go]: make sure this error message won't leak password related information.

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**